packages/slice-machine/src/features/customTypes/customTypesBuilder/CustomTypeProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { CustomType } from "@prismicio/types-internal/lib/customtypes";
+
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import {
+  AutoSaveStatusType,
+  useAutoSave,
+} from "@src/features/autoSave/useAutoSave";
+import { getFormat } from "@src/domain/customType";
+import { updateCustomType } from "@src/apiClient";
+
+import { CUSTOM_TYPES_MESSAGES } from "../customTypesMessages";
+
+type CustomTypeContext = {
+  customType: CustomType;
+  autoSaveStatus: AutoSaveStatusType;
+  setCustomType: Dispatch<SetStateAction<CustomType>>;
+};
+
+type CustomTypeProviderProps = {
+  children: ReactNode | ((value: CustomTypeContext) => ReactNode);
```

Comment:
Yes! Because as soon as we instance it we want to use the automatically updated custom type and autoSave status. And for me having to create a sub component is just more heavy.

See: https://github.com/prismicio/slice-machine/blob/xru/auto-save-type-builder/packages/slice-machine/src/features/customTypes/customTypesBuilder/CustomTypesBuilderPage.tsx#L76